lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -1647,8 +1648,11 @@ def create_default_nic(self):
             if not subnet_id:
                 self.fail(no_subnets_msg)
 
-        self.results['actions'].append('Created default public IP {0}'.format(self.name + '01'))
-        pip = self.create_default_pip(self.resource_group, self.location, self.name + '01', self.public_ip_allocation_method)
+        if self.public_ip_allocation_method != 'Disabled':
```

Comment:
This seems like it could be simplified by changing this to

```
pip = None
if self.public_ip_allocation_method != 'Disabled':
    self.results['actions'].append('Created default public IP {0}'.format(self.name + '01'))
    pip_info = self.create_default_pip(self.resource_group, self.location, self.name + '01', self.public_ip_allocation_method)
    pip = self.network_models.PublicIPAddress(id=pip.id, location=pip.location, resource_guid=pip.resource_guid)
    
....
# now you can just set the model directly instead of putting a ternary operator
parameters.ip_configurations[0].public_ip_address = pip
```